packages/l/libkmod/xmake.lua

```diff
@@ -6,18 +6,130 @@ package("libkmod")
+
+                // from https://android.googlesource.com/kernel/common/+/03c04a7cba972/tools/perf/util/get_current_dir_name.c
+                static inline char *get_current_dir_name(void)
+                {
+                    char pwd[PATH_MAX];
+                    return getcwd(pwd, sizeof(pwd)) == NULL ? NULL : strdup(pwd);
+                }
+            ]], {plain = true})
+        end
+
+        if package:version():lt("v34") then
+            local configs = {
+                "--disable-dependency-tracking",
+                "--disable-manpages",
+                "--disable-test-modules",
+                "--disable-tools"
+            }
+
+            table.insert(configs, "--enable-logging=" .. (package:config("logging") and "yes" or "no"))
+            table.insert(configs, "--with-zstd=" .. (package:config("zstd") and "yes" or "no"))
+            table.insert(configs, "--with-zlib=" .. (package:config("zlib") and "yes" or "no"))
+            table.insert(configs, "--with-xz=" .. (package:config("xz") and "yes" or "no"))
+            table.insert(configs, "--with-openssl=" .. (package:config("openssl") and "yes" or "no"))
+
+            local packagedeps = {}
+            for _, dep in ipairs(package:librarydeps()) do
+                table.insert(packagedeps, dep:name())
+            end
+
+            io.replace("Makefile.am", [[dist_bashcompletion_DATA = \
+	shell-completion/bash/kmod]], "", {plain = true})
+
+            import("package.tools.autoconf").configure(package, configs, {packagedeps = packagedeps})
+
+            io.replace("libtool", "      -all-static)", [[      --target=* )
+        func_append compiler_flags " $arg"
+        func_append linker_flags " $arg"
+        ;;
+      -all-static)]], {plain = true})
+
+            os.vrunv("make")
+            os.vrunv("make install")
+        else
+            local configs = {
+                "-Dbashcompletiondir=no",
+                "-Dfishcompletiondir=no",
+                "-Dzshcompletiondir=no",
+                "-Dmanpages=false",
+                "-Dtools=false"
+            }
+
+            table.insert(configs, "-Dlogging=" .. (package:config("logging") and "true" or "false"))
+
+            table.insert(configs, "-Dzstd=" .. (package:config("zstd") and "enabled" or "disabled"))
+            table.insert(configs, "-Dzlib=" .. (package:config("zlib") and "enabled" or "disabled"))
+            table.insert(configs, "-Dxz=" .. (package:config("xz") and "enabled" or "disabled"))
+            table.insert(configs, "-Dopenssl=" .. (package:config("openssl") and "enabled" or "disabled"))
+
+            io.replace("meson.build", "_tools = %[(.-)%]", "_tools = []")
+
+            import("package.tools.meson").install(package, configs)
+        end
     end)
 
     on_test(function (package)
-        assert(package:has_cfuncs("kmod_new", {includes = "libkmod/libkmod.h"}))
+        assert(package:has_cfuncs("kmod_new", {includes = "libkmod.h"}))
```

Comment:
it will break some packages and user projects.

https://github.com/xmake-io/xmake-repo/blob/28ce132416be2f779a37439fde1195b2a5cd6734/packages/e/eudev/xmake.lua#L28

Maybe we need to set add_includedirs to be compatible with both paths.